### ISSUE TITLE
Diagram from URL does not load

### ISSUE THREAD

**Issue by ben-turner** [jdbranham/grafana-diagram#75]
When using a diagram URL, the error `_this.substituteHashPrefixedNotation is not a function` is thrown. (Full stack trace below.) This causes the diagram to never be displayed. If you visit the URL manually and copy-paste the text into the content field, it loads fine. As per the documentation, the URL returns a text/plain string.

I'm running version 1.4.4, cloned directly from this repo. (commit: 440689793ab6da82019c5ee43b49438dfef976d5)

Stack Trace:
```
TypeError: _this.substituteHashPrefixedNotation is not a function (vendor.2305a8e1d478628b1297.js:43)
    at DiagramCtrl.updateDiagram_cont (diagramControl.js:292)
    at successCallback (diagramControl.js:317)
    at u (vendor.2305a8e1d478628b1297.js:43)
    at vendor.2305a8e1d478628b1297.js:43
    at f.$digest (vendor.2305a8e1d478628b1297.js:43)
    at f.$apply (vendor.2305a8e1d478628b1297.js:43)
    at vendor.2305a8e1d478628b1297.js:43
    at i (vendor.2305a8e1d478628b1297.js:43)
    at vendor.2305a8e1d478628b1297.js:43 "Possibly unhandled rejection: {}"
```

**Comment by ben-turner** [comment]
I just tried using the `test/x-mermaid` content-type as per #59 to no avail.

**Comment by TimEvens** [comment]
I'm running into the same issue.  Interesting enough, the below change resolves the issue. 

```diff

**Comment by briangann** [comment]
I'll work on a fix for this - both methods need to work. The nested function call can be pulled out so the _this reference works correctly.

**Comment by carolbethperkins** [comment]
I hit this issue as well. 

I tried Tim's solution. When I commented out 'var _this' I got an error message: "_this is not defined". As Brian said this is needed in the methods.

Any progress on a fix for this? A solution would be greatly appreciated.

**Comment by briangann** [comment]
I have the fix ready and tested with remote urls and local diagrams.  I noticed the remote diagrams was not populating the node colors, that's been fixed also.

A cache mechanism would be useful for the URL method too to prevent pulling the mermaid file on every refresh.

**Comment by ben-turner** [comment]
I think if a cache was added it would need to be configurable. My use case for using a URL is that the diagram layout changes regularly. A cache would make it no better than a hard-coded diagram.

Something like what grafana does with template vars could be okay though
Refresh:
- Never
- On Dashboard Load
- On Time Range Change

**Comment by fernandobohrer** [comment]
Hi there!

I know it's a little out of topic but what service are you using to generate the mermaid response?

I'm looking for a service but didn't find anything so far. Could you share some extra information about the service itself? Is it like an opensource software that generates the response or have you guys developed something by yourselves?

Thanks in advance,

Fernando

**Comment by nickman** [comment]
Any ETA on this ? Would love to have it working.

**Comment by ben-turner** [comment]
I think we're just waiting on #79 to get merged

**Comment by nickman** [comment]
Seems like it works (#79). I dropped it in my Grafana install and it's working.
I do notice quite a bit of flashing when I have a lot of nodes. I think that was mentioned somewhere as well.

**Comment by nunziox** [comment]
Is there any reason why #79 has not been merged yet ?

**Comment by smile18** [comment]
I am also getting same issue, any ETA on it?

**Comment by fyipal** [comment]
@jdbranham any plans for the merge?
Still, see issues when using Diagrams with URL "Parse error on line 1:"

**Comment by adedommelin** [comment]
@jdbranham any update on this one ?

Thanks !

**Comment by jdbranham** [comment]
Hi @adedommelin and @fyipal 
Which version of the plugin are you using?

**Comment by bale836** [comment]
i use v1.6.1
the URL cannot load too
i cannot get any error from console or grafana logs

**Comment by ferdi-ozdemir** [comment]
Hi,

I am just new with grafana and your diagram.. i faced the same issue with the latest version 1.6.1 with grafana v6.6.1.

Thanks

**Comment by shankarsg** [comment]
I am also facing same issue, URL is not loading.. Grafana v6.5.3 , Diagram 1.6.1

**Comment by nthanes** [comment]
Not sure if related, but will leave this here for anyone else that may encounter this.

It looks like there is a bug when you're using the `url` section, if you don't have any graphs defined in the `content` section, then it fails to load and render the contents from the `url`. Once I added some dummy data into the `content` section, then switched back to `url` it works fine.

Plugin Version: 1.6.2, Grafana Version: 6.5.3

**Comment by jdbranham** [comment]
Sorry guys - I thought this was resolved in the latest version.
I'll take a look.

**Comment by cis2131** [comment]
Is there any way to get this to work?. 
I am using latest version of grafana, and 1.6.2 of grafana-diagram.

**Comment by jdbranham** [comment]
Plugin v1.7+ has better handling of url vs static definition - let's open a new issue if there are further issues.  
Thanks!
